### DISCUSSION TITLE
FormBuilder: TextInput `->required()` works but `->rules('required')` fails

### DISCUSSION THREAD

**Discussion by pekka** [filamentphp/filament#7048] [Help] [answered]
I have a Livewire form in the frontend that is using form builder. I create the form in the component's `getFormSchema()` method. 

Among other things, it returns a `TextInput`:

`TextInput::make('name')->label("Name");`

for logistical reasons, I need to import the validation rules for this field from an array in a model. The array contains the rule `required`. However when I do this:

`TextInput::make('name')->label("Name")->rules([Feedback::$validate_rules_anonymous["name"]])`

the field does not show up as required. Nor does applying the validation rule directly:

`TextInput::make('name')->label("Name")->rules('required')`
`TextInput::make('name')->label("Name")->rules(['required'])`

only when I use `->required()` 

`TextInput::make('name')->label("Name")->required()`

it works.

The [documentation](https://filamentphp.com/docs/2.x/forms/validation#getting-started) says:

> Beware that some validations rely on the field name and therefore won't work when passed via ->rule()/->rules(). Use the dedicated validation methods whenever you can.

But surely that won't apply to something as basic as `required`. Or would it?

**Comment by pboivin** [comment, marked as answer]
The `required()` method also sets the `$isRequired` property on the field. This property is used in the Blade views to handle the front-end validation and required indicator.

**Reply by pekka** [reply]
I see I see. Then I'll have to do that manually, no way around it. Thanks!
